### ISSUE TITLE
Support for PMML or ONNX.

### ISSUE THREAD

**Issue by RakshithGB** [alan-turing-institute/sktime#1240]
**Possible to export the model into PMML or ONNX?**
Having trouble training on 1 architecture (x86) and using it for production on another architecture (arm), would like some utilities to have model persistence across architecture like how scikit-learn provides.

**Ideal solution would be to support both PMML and ONNX**
ONNX seems to have some limitation on certain platforms, PMML seems limited to specific model types only. Would be nice to have support for both.

**Comment by mloning** [comment]
Hi @RakshithGB thanks for the suggestion, sounds like a good idea to me!

**Comment by fkiraly** [comment]
So, @RakshithGB, @mloning, I'm not sure whether this is merely a question of implementation.

PMML's [time series model specification at v4.4](http://dmg.org/pmml/v4-4/TimeSeriesModel.html) seems to cover only forecasting (univariate only?), which would exclude the other tasks; metrics/evaluation is also unclear whether it directly translates.

So before we can implement this for most of `sktime`, some work may have to be done to see how much PMML already covers, and where PMML has to be extended into parts of the learning task taxonomies that we have.

It should be implementable for some of the more classical forecasting models though.

Regarding ONNX, this would have to be extended to time series related learning tasks too, is that correct?
It should be substantially easier though, since explicit `sklearn` interfaces exist, see `sklearn-onnx`, which could be easily extended to cover `sktime`, because `sktime` uses `sklearn` design principles and is interface compatible.

**Comment by RakshithGB** [comment]
@mloning , @fkiraly PMML would surely require a larger effort. I also have one more suggestion, a lot of folks out there are looking at json like serialisation and deserialisation including me. As you can see [here](https://github.com/mlrequest/sklearn-json) much of the standard scikit models have this support, this becomes makes things easier to deploy across even 32 and 64 bit MCUs.

**Comment by fkiraly** [comment]
Hm, @RakshithGB, sounds like this needs some specalist experience - would you be interested to contribute this, since you seem to have some background in the area?

**Comment by RakshithGB** [comment]
@fkiraly I lack bandwidth to work on this. But will surely contribute as time permits .

**Comment by fkiraly** [comment]
Thanks! Let's leave this open then until this issue is found by the right person 😄

**Comment by viggotw** [comment]
Any news on this? :)

**Comment by fkiraly** [comment]
No one has picked this up afaik

**Comment by ojuschugh1** [comment]
Hi @fkiraly , I hope you are doing well. Could you kindly please assign this issue to me, I am interested to work on this issue?
